chromium_src/net/url_request/url_request_http_job.cc

```diff
@@ -30,25 +32,55 @@ bool ShouldUseEphemeralStorage(net::URLRequestHttpJob* http_job) {
 
 }  // namespace
 
+bool URLRequestHttpJob::CanSetCookieIncludingEphemeral(
+    const net::CanonicalCookie& cookie,
+    CookieOptions* options) {
+  return CanUseEphemeralStorage(this) ||
+         CanSetNonEphemeralCookie(cookie, options);
+}
+
+bool URLRequestHttpJob::CanGetNonEphemeralCookies() {
+  // We cannot call CanGetCookies without first checking the privacy mode.
+  return request_info_.privacy_mode == PRIVACY_MODE_DISABLED && CanGetCookies();
+}
+
+bool URLRequestHttpJob::CanSetNonEphemeralCookie(
+    const net::CanonicalCookie& cookie,
+    CookieOptions* options) {
+  return CanSetCookie(cookie, options);
+}
+
+}  // namespace net
+
 #define BRAVE_ADDCOOKIEHEADERANDSTART                                          \
-  if (ShouldUseEphemeralStorage(this))                                         \
+  if (!CanGetNonEphemeralCookies()) {                                          \
+    DCHECK(request()->isolation_info().top_frame_origin().has_value());        \
```

Comment:
I was asked to remove this DHCECK and the curly braces in the original review for this change, so it's probably best to remove them here as well.

Reply:
not remove, just move

Reply:
it would be much easier to  manage these changes if you changed the base branch for the PR

Reply:
I see. Will restore it.